packages/opencensus-instrumentation-mongodb/test/test-mongodb.ts

```
@@ -0,0 +1,280 @@
+    mongodb.MongoClient.connect(url, function connectedClient(err, client) {
+      if (err) {
+        reject(err);
+        return;
+      }
+      const db = client.db(dbName);
+      const collection = db.collection(collectionName);
+      resolve({'client': client, 'collection': collection} as MongoDBAccess);
+    });
+  });
+}
+
+function assertSpan(
+    rootSpanVerifier: RootSpanVerifier, expectedName: string,
+    expectedType: string) {
+  assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 1);
+  assert.strictEqual(rootSpanVerifier.endedRootSpans[0].spans.length, 1);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].name, expectedName);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].type, expectedType);
+}
+
+describe('MongoDBPlugin', () => {
+  // For these tests, mongo must be runing. Add OPENCENSUS_MONGODB_TESTS to run
+  // these tests.
+  const OPENCENSUS_NETWORK_TESTS =
+      process.env.OPENCENSUS_NETWORK_TESTS as string;
+  let shouldTest = true;
+  if (!OPENCENSUS_NETWORK_TESTS) {
+    console.log('Skipping test-mongodb. Run MongoDB to test');
+    shouldTest = false;
+  }
+
+  const url = 'mongodb://localhost:27017';
+  const dbName = 'opencensus-tests';
+  const collectionName = 'test';
+  const VERSION = process.versions.node;
+
+  const tracer = new classes.Tracer();
+  const rootSpanVerifier = new RootSpanVerifier();
+  let client: mongodb.MongoClient;
+  let collection: mongodb.Collection;
+
+  before((done) => {
+    tracer.start({samplingRate: 1});
+    tracer.registerSpanEventListener(rootSpanVerifier);
+    plugin.applyPatch(mongodb, tracer, VERSION);
+    accessCollection(url, dbName, collectionName)
+        // tslint:disable-next-line:no-any
+        .then((result: any) => {
+          client = result.client;
+          collection = result.collection;
+          done();
+        })
+        .catch((err: Error) => {
+          console.log(
```

Shouldn't an error condition fail this test rather than allowing it to skip?

---

I made it skip so it doesn't block the other tests in case the user doesn't have mongo running. I feel like failing here implies that there is a problem with the plugin behavior rather than with the environment.